selfdrive/controls/lib/lateral_planner.py

```diff
@@ -79,6 +79,9 @@ def update(self, sm):
                      y_pts,
                      heading_pts)
     # init state for next
+    # mpc.u_sol is the desired curvature rate from x0[3]. 
+    # we integrate here so that x_sol[:,3] is the desired curvature for lat_control
+    # if we set x0[3] = measured_curvature the output would be desired_curvature_rate
```

Comment:
I don't understand this whole comment block. Integrate or Interpolate.

Reply:
Wanted to clarify it's interpolating over the integral of rate, since rate is really the MPC's "output" state. 

This is better maybe:

# mpc.u_sol is the desired curvature rate given x0 curv state. 
# with x0[3] = measured_curvature, this would be the actual desired rate.
# instead, interpolate x_sol so that x0[3] is the desired curvature for lat_control.

Reply:
Yeah that makes sense!